imports/plugins/custom/order-cancelation/client/components/cancelableCompletedShopOrder.js

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { Meteor } from "meteor/meteor";
+import PropTypes from "prop-types";
+import { Components, replaceComponent } from "@reactioncommerce/reaction-components";
+import Modal from "./Modal";
+
+/**
+ * @summary Displays the order breakdown for each Shop
+ * @param {Object} props - React PropTypes
+ * @property {String} shopName - The name of the shop
+ * @property {Array} items - an array of individual items for this shop
+ * @property {Function} handleDisplayMedia - A function for displaying product images
+ * @property {boolean} isProfilePage - Checks if current page is profile page
+ * @return {Node} React node containing the break down of the order by Shop
+ */
+
+class CancelableCompletedShopOrder extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      showCancelModal: false
+    };
+
+    this.toggleCancelModal = this.toggleCancelModal.bind(this);
+    this.cancelOrder = this.cancelOrder.bind(this);
+  }
+
+  toggleCancelModal(event) {
+    event.preventDefault();
+    this.setState({
+      showCancelModal: !this.state.showCancelModal
+    });
```

Comment:
Using setState with values that are currently in state like you did with `!this.state.showCancelModal` can be buggy because `setState` is an async method and repeated fast calls to the method may not behave the way we expect it to. To solve this you can use the callback method of setting state instead by passing a callback to the `setState` method this callback receives the previous state as the first argument and should return the new state object removing the need to use `this.state`. Based on this, the  `setState` in the `toggleCancelModal` can be rewritten as 
```
 this.setState(state => ({ 
      showCancelModal: !state.showCancelModal 
})
```

Reply:
Thanks for the snippet. I'll implement it where necessary.